src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -13383,24 +13383,18 @@ private ParameterSyntax ParseLambdaParameter()
             var identifier = this.ParseIdentifierToken();
             ParseParameterNullCheck(ref identifier, out var equalsToken);
 
-            // If we didn't already consume an equals sign as part of !!=, then try to scan one out now. Note: this is
-            // not legal code.  But we detect it so that we can give the user a good message, and so we don't go
-            // completely off the rails.
-            //
-            // Note: we add the `= value` as skipped trivia to either the identifier or `!!` (if we have the latter).
-            // This allows us to handle this code without ever showing it the binding phases.  We could consider
-            // actually binding this in the future if it would be helpful and if we're ok paying the testing cost of
-            // checking this at the semantic layers.
-            equalsToken ??= TryEatToken(SyntaxKind.EqualsToken);
 
-            if (equalsToken != null)
-            {
-                equalsToken = AddError(equalsToken, ErrorCode.ERR_DefaultValueNotAllowed);
+            // Handle the case where we have a default (optional) lambda parameter by eating '=', followed by 
+            // an EqualsValueClause which is simply an expression.
+            // The EqualsValueClause is then added as an attribute of the
+            // parsed parameter
```

Comment:
Minor: The comment seems unnecessary to me. It seems to imply something the reader should pay attention to, but we're not doing anything unusual here.

Consider removing the comment or replacing with a simpler comment such as: "// Parse default value if any."